infrastructures/docker/src/main/java/org/eclipse/che/workspace/infrastructure/docker/provisioner/installer/InstallerConfigApplier.java

```diff
@@ -83,11 +74,10 @@ public void apply(Environment envConfig, DockerEnvironment dockerEnvironment)
    * @param machine affected machine
    * @throws InstallerException if any error occurs
    */
-  public void apply(@Nullable MachineConfig machineConf, DockerContainerConfig machine)
+  public void apply(@Nullable InternalMachineConfig machineConf, DockerContainerConfig machine)
       throws InstallerException {
     if (machineConf != null) {
-      for (Installer installer :
-          installerRegistry.getOrderedInstallers(machineConf.getInstallers())) {
+      for (Installer installer : machineConf.getInstallers()) {
```

Comment:
`InstallerRegistry.getOrderedInstallers` also includes dependencies in the resulted list.
So, in case of `machineConf.getInstallers()` a configuration of some dependent installers won't be applied.